stdlib/public/core/Character.swift

```diff
@@ -203,13 +202,12 @@ public struct Character :
   @_versioned
   internal init(_largeRepresentationString s: String) {
     if let native = s._core.nativeBuffer,
-       native.start == s._core._baseAddress! {
+      native.start == s._core._baseAddress!,
+      native.usedCount == s._core.count {
       _representation = .large(native._storage)
       return
     }
-    var nativeString = ""
-    nativeString.append(s)
-    _representation = .large(nativeString._core.nativeBuffer!._storage)
+    _representation = .large(_StringCore(s.utf16).nativeBuffer!._storage)
```

Comment:
It's already guaranteed to be one character, based on usage.  But a _sanityCheck would be good.